contrib/platform/test/com/sun/jna/platform/win32/Shell32Test.java

```diff
@@ -232,4 +233,18 @@ public void testExtractIconEx() {
         int iconCount = Shell32.INSTANCE.ExtractIconEx(new File(winDir, "explorer.exe").getAbsolutePath(), -1, null, null, 1);
         assertTrue("Should be at least two icons in explorer.exe", iconCount > 1);
     }
+
+    public void testCurrentProcessExplicitAppUserModelID() {
+        String appUserModelID = "com.sun.jna.platform.win32.Shell32Test";
+
+        HRESULT r1 = Shell32.INSTANCE.SetCurrentProcessExplicitAppUserModelID(new WString(appUserModelID));
+        assertEquals(WinError.S_OK, r1);
+
+        PointerByReference ppszAppID = new PointerByReference();
+        HRESULT r2 = Shell32.INSTANCE.GetCurrentProcessExplicitAppUserModelID(ppszAppID);
+        assertEquals(WinError.S_OK, r2);
+
+        assertEquals(appUserModelID, ppszAppID.getPointer().getWideString(0));
```

Comment:
Could you please free the string, as this is expected from the documentation? For the unittest this is not actually necessary, but people looking at the code might be mislead, that you don't need to free the string.